api/src/main/java/org/openmrs/PatientProgram.java

```diff
@@ -36,23 +53,55 @@ public class PatientProgram extends BaseChangeableOpenmrsData implements Customi
 	// ******************
 	// Properties
 	// ******************
-	
+	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "")
+	@GenericGenerator(
+			name = "patient_program_id_seq",
+			strategy = "native",
+			parameters = @Parameter(name = "sequence", value = "patient_program_patient_program_id_seq")
+			)
+	@Column(name = "patient_program_id")
```

Comment:
The current configuration uses a `sequence` strategy, but the XML mapping uses `identity-based` key generation. Please update the strategy to `GenerationType.IDENTITY` to match the XML